samples/openapi3/client/petstore/python-experimental/test/test_drawing.py

```diff
@@ -0,0 +1,58 @@
+# coding: utf-8
+
+"""
+    OpenAPI Petstore
+
+    This spec is mainly for testing Petstore server and contains fake endpoints, models. Please do not use this for any other purpose. Special characters: \" \\  # noqa: E501
+
+    The version of the OpenAPI document: 1.0.0
+    Generated by: https://openapi-generator.tech
+"""
+
+
+from __future__ import absolute_import
+
+import unittest
+
+import petstore_api
+
+
+class TestDrawing(unittest.TestCase):
+    """Drawing unit test stubs"""
+
+    def setUp(self):
+        self.api_client = petstore_api.ApiClient()
+
+    def tearDown(self):
+        pass
+
+    def test_deserialize_oneof_reference(self):
+        isosceles_triangle = petstore_api.Shape(
+            shape_type="Triangle",
+            triangle_type="IsoscelesTriangle"
+        )
+        assert isinstance(isosceles_triangle, petstore_api.IsoscelesTriangle)
+        inst = petstore_api.Drawing(
+            # 'main_shape' has type 'Shape', which is a oneOf [triangle, quadrilateral]
+            # composed schema. So we should be able to assign a petstore_api.Triangle
+            # to a 'main_shape'.
+            main_shape=isosceles_triangle,
+            shapes=[
+                petstore_api.Shape(
+                    shape_type="Triangle",
+                    triangle_type="EquilateralTriangle"
+                ),
+                petstore_api.Shape(
```

Comment:
Can we keep one of these as a Triangle, or a specific shape like EquilateralTriangle?
That requires that our solution depends upon shape's discriminator and not on instance._visited_composed_classes containing Shape.

Reply:
I've improved the unit tests.